domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -1483,6 +1483,57 @@ public void getByIdAsString(TestContext context) {
     });
   }
 
+  @Test
+  public void getByIdAsStringForUpdate(TestContext context) {
+    Async async1 = context.async();
+    Async async2 = context.async();
+    String id = randomUuid();
+    postgresClient = createFoo(context);
+    postgresClient.save(FOO, id, new StringPojo("x0"), res -> {
+      assertSuccess(context, res);
+      context.assertEquals(id, res.result());
+      Criterion criterion = new Criterion();
+      criterion.addCriterion(new Criteria().addField("'id'").setOperation("=").setVal(id));
+      postgresClient.startTx(conn -> {
+        // lock the row for update
+        postgresClient.getByIdAsStringForUpdate(conn, FOO, id, get -> {
+          // concurrent update will have to wait
+          postgresClient.update(FOO, new StringPojo("x2"), id, rows -> {
+            context.assertTrue(async1.isCompleted());
+            postgresClient.getByIdAsString(FOO, id, get2 -> {
+              assertSuccess(context, get2);
+              context.assertTrue(get2.result().contains("\"x2\""));
+              async2.complete();
+            });
+          });
+          // update first because it holds the lock
+          postgresClient.execute(conn, "SELECT pg_sleep(0.5)", delayRs -> {
```

Comment:
I don't think that we need the pg_sleep. Instead we can simply run "SELECT 1" to make sure that the update x2 has reached the database and is blocked before we run update x1.

Reply:
Can you clarify how to use `SELECT 1` to make sure that the update x2 has reached the db? We could query system tables to check locks (if that is what you mean) but it is probably overkill than `pg_sleep`.